### PR TITLE
[codex] Remove production-usable default admin credentials

### LINKED ISSUE #7475: Remove production-usable default admin credentials and API keys
## Source

Derived from `doc/security-assessment/elsa-workflows-elsa-core-findings-2026-05-20.md` and `doc/security-assessment/elsa-core-software-quality.md`.

## Problem

The assessment identifies multiple production-usable default credential paths in the identity module and reference server configuration:

- `AdminApiKeyProvider` grants admin claims for the all-zero GUID API key.
- `AdminUserProvider` returns an admin user for the static password `password` regardless of the supplied username filter.
- `src/apps/Elsa.Server.Web/appsettings.json` includes seeded users with known credentials suitable for demo use but unsafe as production defaults.

These should be treated as explicit demo/bootstrap behavior, never as silent production defaults.

## Action

- Remove static admin fallback credentials from default provider wiring.
- Require explicit operator-supplied bootstrap credentials or generate one-time credentials at first start.
- Fail fast when known demo credentials are present outside an explicit demo/development profile.
- Add tests proving the all-zero API key and arbitrary username plus `password` do not authenticate by default.
- Document the safe bootstrap flow for local development and production deployments.

## Acceptance criteria

- Fresh production-style configuration does not accept any hard-coded admin API key or password.
- Demo credentials require an explicit demo/development opt-in.
- Upgrade notes explain any migration impact for sample apps and Docker images.

### PR DESCRIPTION
**Pull request by sfmskywalker**
Fixes #7475

## Summary
- require explicit admin username/password/API key configuration outside development
- move default admin credentials behind development-only behavior and document the configuration
- cover the startup and provider behavior with identity tests

## Validation
- dotnet test test/unit/Elsa.Identity.UnitTests/Elsa.Identity.UnitTests.csproj --no-restore
- jq empty src/apps/Elsa.Server.Web/appsettings.json src/apps/Elsa.Server.Web/appsettings.Development.json src/apps/Elsa.Server.Web/appsettings.Production.json
- dotnet build test/component/Elsa.Workflows.ComponentTests/Elsa.Workflows.ComponentTests.csproj --no-restore

### CODE REVIEW

**Comment by sfmskywalker** [comment]
@greptile review

@copilot review

**Comment by sfmskywalker** [comment]
@greptile review

File: src/modules/Elsa.Identity/ShellFeatures/DefaultAuthenticationFeature.cs [hunk 1]
```diff
@@ -27,10 +29,37 @@ public class DefaultAuthenticationFeature : IShellFeature
...
+    public bool UseDevelopmentAdminApiKey { get; set; }
+
     public void ConfigureServices(IServiceCollection services)
     {
         services.ConfigureOptions<ConfigureJwtBearerOptions>();
         services.AddIdentityTokenOptionsValidation();
+        services.Configure<AdminApiKeyOptions>(options =>
+        {
+            options.ApiKey = UseDevelopmentAdminApiKey ? AdminApiKeyProvider.DevelopmentApiKey : AdminApiKey;
+        });
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Identity/ShellFeatures/DefaultAuthenticationFeature.cs:67]: Resolved by switching the shell feature to AdminApiKeyProvider whenever an explicit or development admin API key is configured, before authentication is wired.

[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Identity/ShellFeatures/DefaultAuthenticationFeature.cs:67]: Resolved by switching the shell feature to AdminApiKeyProvider whenever an explicit or development admin API key is configured, before authentication is wired.

File: src/apps/Elsa.Server.Web/appsettings.json [hunk 2]
```diff
@@ -81,54 +81,6 @@
...
-        "ClientSecret": "a<~QGGHTEA%u4;CU&'Wga5ED:_&Gd1C)",
-        "HashedApiKey": "Z5ClHs3mbzx8Pnw3+PxbMq8A/Y+VKMCCDTGYtax8JFM=",
-        "HashedApiKeySalt": "kBisa1X8FwBfN2zmyGMFRgIVVBleghhQAJ4WGyTkaD0=",
-        "HashedClientSecret": "jEv58d0SVbGQ3nBZM0lkzHghG4Y+lMKW80wipz+9vHk=",
-        "HashedClientSecretSalt": "xRKy14Ok1/tU3kLf/8V1fcbLIegy9vcM90Peu2tzohU=",
-        "TenantId": "default"
-      }
     ]
   },
   "Scripting": {
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/apps/Elsa.Server.Web/appsettings.json:86]: Resolved by adding Elsa.Server.Web appsettings.Development.json with the development-only identity users/applications and signing key, leaving appsettings.json without hardcoded development credentials.
